commands/monitoring.go

```diff
@@ -0,0 +1,350 @@
+	AddStringSliceFlag(cmdAlertPolicyUpdate, doctl.ArgAlertPolicyEntities, "", nil, "Entities to apply the alert against.")
+	AddStringSliceFlag(cmdAlertPolicyUpdate, doctl.ArgAlertPolicySlackChannels, "", nil, "Slack channels to send alerts to.")
+	AddStringSliceFlag(cmdAlertPolicyUpdate, doctl.ArgAlertPolicySlackURLs, "", nil, "Slack URLs to send alerts to.")
+
+	CmdBuilder(cmd, RunCmdAlertPolicyGet, "get <alert-policy-uuid>", "Retrieve information about an alert policy", `Use this command to retrieve an alert policy and see its configuration.`, Writer,
+		displayerType(&displayers.AlertPolicy{}))
+
+	CmdBuilder(cmd, RunCmdAlertPolicyList, "list", "List all alert policies", `Use this command to retrieve a list of all the alert policies in your account.`, Writer,
+		aliasOpt("ls"), displayerType(&displayers.AlertPolicy{}))
+
+	cmdRunAlertPolicyDelete := CmdBuilder(cmd, RunCmdAlertPolicyDelete, "delete <alert-policy-uuid>...", "Delete an alert policy", `Use this command to delete an alert policy.`, Writer)
+	AddBoolFlag(cmdRunAlertPolicyDelete, doctl.ArgForce, doctl.ArgShortForce, false, "Delete an alert policy without confirmation prompt")
+
+	return cmd
+}
+
+// RunCmdAlertPolicyCreate runs alert policy create.
+func RunCmdAlertPolicyCreate(c *CmdConfig) error {
+	ms := c.Monitoring()
+
+	desc, err := c.Doit.GetString(c.NS, doctl.ArgAlertPolicyDescription)
+	if err != nil {
+		return err
+	}
+	alertType, err := c.Doit.GetString(c.NS, doctl.ArgAlertPolicyType)
+	if err != nil {
+		return err
+	}
+	value, err := c.Doit.GetInt(c.NS, doctl.ArgAlertPolicyValue)
+	if err != nil {
+		return err
+	}
+
+	window, err := c.Doit.GetString(c.NS, doctl.ArgAlertPolicyWindow)
+	if err != nil {
+		return err
+	}
+
+	entities, err := c.Doit.GetStringSlice(c.NS, doctl.ArgAlertPolicyEntities)
+	if err != nil {
+		return err
+	}
+
+	tags, err := c.Doit.GetStringSlice(c.NS, doctl.ArgAlertPolicyTags)
+	if err != nil {
+		return err
+	}
+
+	enabled, err := c.Doit.GetBoolPtr(c.NS, doctl.ArgAlertPolicyEnabled)
+	if err != nil {
+		return err
+	}
+
+	compareStr, err := c.Doit.GetString(c.NS, doctl.ArgAlertPolicyCompare)
+	if err != nil {
+		return err
+	}
+
+	compare, err := getCompartor(compareStr)
+	if err != nil {
+		return err
+	}
+
+	emails, err := c.Doit.GetStringSlice(c.NS, doctl.ArgAlertPolicyEmails)
+	if err != nil {
+		return err
+	}
+
+	slackChannels, err := c.Doit.GetStringSlice(c.NS, doctl.ArgAlertPolicySlackChannels)
+	if err != nil {
+		return err
+	}
+
+	slackURLs, err := c.Doit.GetStringSlice(c.NS, doctl.ArgAlertPolicySlackURLs)
+	if err != nil {
+		return err
+	}
+
+	if len(slackURLs) != len(slackChannels) {
+		return errors.New("must provide the same number of slack channels as slack URLs")
+	}
+
+	slacks := make([]godo.SlackDetails, len(slackChannels))
+	for i, channel := range slackChannels {
+		slacks[i] = godo.SlackDetails{Channel: channel, URL: slackURLs[i]}
+	}
+
+	apcr := &godo.AlertPolicyCreateRequest{
```

Comment:
Is there any other input validation we should be doing before making the request?

Reply:
I went ahead and added validation on the actual window supplied, the alert policy type, and that there are emails and/or slacks supplied!

Reply:
I did that for both create and update.

Reply:
I added some more input validation for both the update and create commands!